### ISSUE TITLE
question about the $z^e$ in equ.9

### ISSUE THREAD

**Issue by dsj96** [twinkle0331/LGTM#3]
This work is very interesting. In section 2 Notations, define $z^e$ as the batch of validation  samples.
What I am curious about is how to select the validation batch for each training sample to calculate the gradient similarity between the training sample $z^{r}_{i}$ and the validation batch $z^e$.  Different selection of $z^e$ may result in significant differences in gradient similarity.  I don't know if I misunderstood.
Looking forward to your reply ！: ）

**Comment by twinkle0331** [comment]
$z^e$ is randomly choosen from the validation set. Similar to the concept of SGD, here we sample a batch from the validation set.The selection of $z^e$ should be random to represent the performance on unseen dataset.  For each training step, the $z^e$ sampling from the validation set will be different. As the training process proceeds, the validation batch will iterate over all samples in the validation set.

**Comment by twinkle0331** [comment]
For more details, you can refer to our code https://github.com/twinkle0331/LGTM/blob/main/run_glue.py#L689

**Comment by dsj96** [comment]
Thank you!
